rust/src/lib.rs

```diff
@@ -2493,6 +2493,13 @@ impl FfiBoxedSlice {
         let slim_ptr: *mut u8 = fat_ptr as _;
         Box::into_raw(Box::new(FfiBoxedSlice { ptr: slim_ptr, len }))
     }
+
+    fn none() -> *mut Self {
+        Box::into_raw(Box::new(Self {
+            ptr: ptr::null_mut(),
+            len: 0,
+        }))
+    }
```

Comment:
I meant to disagree with changing `FfiBoxedSlice` to allow wrapping a null pointer, but I don't think I got around to submitting that comment :-/ If anything I think we should do the opposite: explicitly disallow null pointers. We don't need to use `Ok(FfiBoxedSlice::none())`; we can use an error instead.